src/lib.rs

```diff
@@ -295,33 +296,34 @@ impl Default for TrackType {
 }
 
 /// The media's global (mvhd) timescale.
-#[derive(Debug, Copy, Clone)]
-struct MediaTimeScale(u64);
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct MediaTimeScale(pub u64);
 
 /// A time scaled by the media's global (mvhd) timescale.
-#[derive(Debug, Copy, Clone)]
-struct MediaScaledTime(u64);
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct MediaScaledTime(pub u64);
 
 /// The track's local (mdhd) timescale.
-#[derive(Debug, Copy, Clone)]
-struct TrackTimeScale(u64, usize);
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct TrackTimeScale(pub u64, pub usize);
 
 /// A time scaled by the track's local (mdhd) timescale.
-#[derive(Debug, Copy, Clone)]
-struct TrackScaledTime(u64, usize);
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct TrackScaledTime(pub u64, pub usize);
 
 #[derive(Debug, Default)]
-struct Track {
+pub struct Track {
     id: usize,
-    track_type: TrackType,
-    empty_duration: Option<MediaScaledTime>,
-    media_time: Option<TrackScaledTime>,
-    timescale: Option<TrackTimeScale>,
-    duration: Option<TrackScaledTime>,
+    pub track_type: TrackType,
+    pub empty_duration: Option<MediaScaledTime>,
+    pub media_time: Option<TrackScaledTime>,
+    pub timescale: Option<TrackTimeScale>,
+    pub duration: Option<TrackScaledTime>,
     track_id: Option<u32>,
-    mime_type: String,
-    data: Option<SampleEntry>,
-    tkhd: Option<TrackHeaderBox>, // TODO(kinetik): find a nicer way to export this.
+    pub mime_type: String,
+    fragmented: (bool, bool, bool), // true when there is no data in stts, stsc, and stco.
```

Comment:
I guess if you want the presence of any sample table data to prevent a track from being marked fragmented, you need to track all three of these. I'd prefer you used a struct with named fields for that. That makes it easier to read and harder to confuse which flag is which without using extra storage.

You could also use a HashSet with enum values (BoxType would serve well) which is higher overhead but makes checking if they're all set a little nicer.